src/components/Main.tsx

```diff
@@ -0,0 +1,687 @@
+        options={{
+          title: s.strings.title_terms_of_service
+        }}
+      />
+
+      <Stack.Screen name="fioAddressList" component={ifLoggedIn(FioAddressListScene)} />
+
+      <Stack.Screen
+        name="fioAddressRegister"
+        component={ifLoggedIn(FioAddressRegisterScene)}
+        options={{
+          headerTitle: () => <EdgeLogoHeader />
+        }}
+      />
+
+      <Stack.Screen
+        name="fioAddressRegisterSelectWallet"
+        component={ifLoggedIn(FioAddressRegisterSelectWalletScene)}
+        options={{
+          title: s.strings.title_fio_address_confirmation,
+          headerRight: () => null
+        }}
+      />
+
+      <Stack.Screen
+        name="fioDomainRegister"
+        component={ifLoggedIn(FioDomainRegisterScene)}
+        options={{
+          headerTitle: () => <EdgeLogoHeader />
+        }}
+      />
+      <Stack.Screen
+        name="fioDomainRegisterSelectWallet"
+        component={ifLoggedIn(FioDomainRegisterSelectWalletScene)}
+        options={{
+          title: s.strings.title_register_fio_domain,
+          headerRight: () => null
+        }}
+      />
+      <Stack.Screen
+        name="fioDomainConfirm"
+        // @ts-expect-error
+        component={ifLoggedIn(FioNameConfirmScene)}
+        options={{
+          headerRight: () => null
+        }}
+      />
+
+      <Stack.Screen
+        name="fioNameConfirm"
+        component={ifLoggedIn(FioNameConfirmScene)}
+        options={{
+          headerRight: () => null
+        }}
+      />
+
+      <Stack.Screen
+        name="fioAddressDetails"
+        component={ifLoggedIn(FioAddressDetailsScene)}
+        options={{
+          headerTitle: () => <ParamHeaderTitle<'fioAddressDetails'> fromParams={params => params.fioAddressName} />
+        }}
+      />
+      <Stack.Screen
+        name="fioConnectToWalletsConfirm"
+        component={ifLoggedIn(FioConnectWalletConfirmScene)}
+        options={{
+          title: s.strings.title_fio_connect_to_wallet
+        }}
+      />
+
+      <Stack.Screen
+        name="fioAddressSettings"
+        component={FioAddressSettingsScene}
+        options={{
+          title: s.strings.title_fio_address_settings
+        }}
+      />
+
+      <Stack.Screen
+        name="fioAddressRegisterSuccess"
+        component={ifLoggedIn(FioAddressRegisteredScene)}
+        options={{
+          headerTitle: () => <ParamHeaderTitle<'fioAddressRegisterSuccess'> fromParams={params => params.fioName} />,
+          headerLeft: () => null
+        }}
+      />
+
+      <Stack.Screen
+        name="fioDomainSettings"
+        component={FioDomainSettingsScene}
+        options={{
+          title: s.strings.title_fio_domain_settings
+        }}
+      />
+
+      <Stack.Screen name="fioRequestList" component={ifLoggedIn(FioRequestListScene)} />
+      <Stack.Screen
+        name="fioRequestApproved"
+        // @ts-expect-error
+        component={ifLoggedIn(TransactionDetailsScene)}
```

Comment:
This is a huge abuse of the transaction details scene, but it's been this way for a while. Let's leave this `// @ts-expect-error` and fix it later.

Reply:
I looked at it, and I would assume we could just navigate to `transactionDetails` key, but then it's missing the `walletId` param; how did this ever work without the `walletId` param. The `FioRequestListScene` is lengthy with many wallet IDs being thrown around, so without much context on that component I wouldn't know which ID is the correct one to pass along. Can we save this for a sidebar and follow up task?

Reply:
Ok. My guess is that this is actually a genuine bug, but the `withWallet` HOC identifies that the `null` wallet doesn't exist, and backs out before we can properly crash. So, the only symptom is that you don't go to the transaction details scene.